src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -748,33 +879,67 @@ public override int GetHashCode()
         /// </summary>
         public static bool operator <(SemanticVersion v1, SemanticVersion v2)
         {
-            if ((object)v1 == null) throw PSTraceSource.NewArgumentException(nameof(v1));
-            return (v1.CompareTo(v2) < 0);
+            return (Compare(v1, v2) < 0);
         }
 
         /// <summary>
         /// Overloaded &lt;= operator
         /// </summary>
         public static bool operator <=(SemanticVersion v1, SemanticVersion v2)
         {
-            if ((object)v1 == null) throw PSTraceSource.NewArgumentException(nameof(v1));
-            return (v1.CompareTo(v2) <= 0);
+            return (Compare(v1, v2) <= 0);
         }
 
         /// <summary>
         /// Overloaded &gt; operator
         /// </summary>
         public static bool operator >(SemanticVersion v1, SemanticVersion v2)
         {
-            return (v2 < v1);
+            return (Compare(v1, v2) > 0);
         }
 
         /// <summary>
         /// Overloaded &gt;= operator
         /// </summary>
         public static bool operator >=(SemanticVersion v1, SemanticVersion v2)
         {
-            return (v2 <= v1);
+            return (Compare(v1, v2) >= 0);
+        }
+
+        private static int ComparePreLabel(string preLabel1, string preLabel2)
+        {
+            // Symver 2.0 standard p.9 - Pre-release versions have a lower precedence than the associated normal version.
+            if (String.IsNullOrEmpty(preLabel1)) { return String.IsNullOrEmpty(preLabel2) ? 0 : 1; }
+            if (String.IsNullOrEmpty(preLabel2)) { return -1; }
+
+            var units1 = preLabel1.Split('.');
+            var units2 = preLabel2.Split('.');
+
+            var minLength = units1.Length < units2.Length ? units1.Length : units2.Length;
+
+            for (int i = 0; i < minLength; i++)
+            {
+                var ac = units1[i];
+                var bc = units2[i];
+                int number1, number2;
+                var isNumber1 = Int32.TryParse(ac, out number1);
+                var isNumber2 = Int32.TryParse(bc, out number2);
+
+                if (isNumber1 && isNumber2)
+                {
+                    if (number1 != number2) { return number1 < number2 ? -1 : 1; }
+                }
+                else
+                {
+                    if (isNumber1) { return -1; }
```

Comment:
It would be great if you can add comment here saying that "Numeric identifiers always have lower precedence than non-numeric identifiers. "

Reply:
I copy-paste all text from the standard.